cbmc_viewer/coveraget.py

```diff
@@ -227,6 +234,46 @@ def merge_coverage_data(coverage_list):
         raise UserWarning("Error merging coverage data: {}".format(error)) from error
     return coverage
 
+def filter_reachable_functions(coverage, reachable_functions=None):
+    """Restrict coverage data to coverage for reachable functions"""
+
+    if not reachable_functions:
+        return coverage
+
+    def valid_coverage_data(coverage, reachable):
+        valid = True
+        for file_name, file_coverage in coverage.items():
+            for func_name, func_coverage in file_coverage.items():
+                if func_name in reachable.get(file_name, {}):
+                    if not func_coverage:
+                        logging.warning("Reachable function %s has no coverage data", func_name)
+                        valid = False
+                else:
+                    lines = [int(line) for line, status in func_coverage.items()
+                             if status is not Status.MISSED]
+                    if lines:
+                        logging.warning(
+                            "Unreachable function %s has reachable lines %s",
+                            func_name, ', '.join([str(line) for line in sorted(lines)])
+                        )
+                        valid = False
+        return valid
+
+    if not valid_coverage_data(coverage, reachable_functions):
+        logging.warning("Coverage calculations may not be accurate")
```

Comment:
What do we do with these warnings? Does the user see them when they view the report?

Reply:
The warnings are printed on the console when we summarize the coverage.  Are you advocating that this be encoded in the json so it will appear on the report?

Reply:
If possible, yes. Otherwise, a user interacting with the report via CI will not have an easy way to know there were warnings.

Reply:
Agree with @danielsn re adding to json.